routes/users.js

```diff
@@ -1,9 +1,100 @@
-const express = require('express');
-const router = express.Router();
+const express = require('express')
+const { User, db } = require('../database/User.js')
+const router = express.Router()
+const mongoose = require('mongoose')
 
-/* GET users listing. */
-router.get('/', (request, response, next) => {
-  response.send('respond with a resource');
-});
 
-module.exports = router;
+router.get('/', ( request, response, next ) => {
+  response.send( 'respond with a resource' )
+})
+
+router.post('/', ( request, response, next ) => {
+  const { username } = request.body
+  const newUser = new User({
+    username: username,
+    dashboards: {
+       widgets: {}
+    }
+  })
+  newUser.save( ( error ) => {
+    if( error ) return handleError( error )
+  })
+})
+
+router.get('/widgets/:widgetId', ( request, response, next ) => {
+  const { widgetId } = request.params
+
+  User.find( {'widgets._id': widgetId},function(error, widgets){
+      return widgets
+  })
+  .then( data => console.log('finished') )
+})
+
+router.post('/:userId/widgets', (request, response, next ) => {
```

Comment:
Widgets do not belong to a user, so it does not seem to make sense to include them here.  I could see an argument being made for `/dashboards/:id/widgets`, but we would probably just return the widgets along with the dashboard - it wouldn't make sense to have a set of widgets outside of a dashboard.  I think this route will also be removed when we clean up the `User` model, as requested above (widgets do not belong to users).